Public/Src/Utilities/Storage/ChangeTracking/FileChangeTracker.cs

```diff
@@ -730,10 +730,53 @@ public Possible<PathExistence> TryProbeAndTrackPath(string path, bool? isReadOnl
 
         private void DisableTracking(string path, Failure trackingFailure)
         {
-            if (Interlocked.Exchange(ref m_trackingStateValue, (int)FileChangeTrackingState.DisabledSinceTrackingIsIncomplete) !=
-                (int)FileChangeTrackingState.DisabledSinceTrackingIsIncomplete)
+            int disabledValue = (int)FileChangeTrackingState.DisabledSinceTrackingIsIncomplete;
+
+            if (!m_volumeMap.SkipTrackingJournalIncapableVolume)
+            {
+                // Immediately disable tracking.
+                DoDisable(path, trackingFailure);
+            }
+            else
             {
-                Logger.Log.DisableChangeTracker(m_loggingContext, path, trackingFailure.DescribeIncludingInnerFailures());
+                // Check if the path is in the volume that can be skipped.
+                // This is expensive, and should only be used during testing.

```

Comment:
Is that mean this block of code will be removed in the future?

Reply:
No it isn't. But the SkipTrackingJournalIncapableVolume should only be set true for testing.

---
In reply to: [293586421](https://github.com/microsoft/BuildXL/pull/457#discussion_r293586421) [](ancestors = 293586421)